app/utils/poller.js

```diff
@@ -0,0 +1,29 @@
+import Ember from 'ember';
+
+export default Ember.Object.extend({
+  interval: function() {
+    return 10000; // Time between polls (in ms)
+  }.property().readOnly(),
+
+  // Schedules the function `f` to be executed every `interval` time.
+  schedule: function(f) {
+    return Ember.run.later(this, function() {
+      f.apply(this);
+      this.set('timer', this.schedule(f));
+    }, this.get('interval'));
+  },
+
+  // Stops the poller
+  stop: function() {
+    Ember.run.cancel(this.get('timer'));
+  },
+
+  // Starts the poller, i.e. executes the `onPoll` function every interval.
+  start: function() {
+    this.set('timer', this.schedule(this.get('onPoll')));
```

Comment:
onPoll is passed once here to schedule so changing it later (with poller.set('onPoll',...)) will continue calling the original function until start() is called again.